### PR TITLE
feat: typescript typings

### PR DESCRIPTION
**Pull request by mimbrown**

### CODE REVIEW

File: src/component.d.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+     */
+    onClose(selectedDates: Date[], dateStr: string): void;
+    /**
+     * Watch for the disabled property and sets the value to the real input.
+     *
+     * @param newState
+     */
+    watchDisabled(newState: boolean): void;
+}, ComponentOptionsMixin, ComponentOptionsMixin, ["blur", "update:modelValue", "on-change", "on-close", "on-destroy", "on-month-change", "on-open", "on-year-change", "on-value-update", "on-day-create", "on-parse-config", "on-ready", "on-pre-calendar-position", "on-key-down"], "blur" | "update:modelValue" | "on-change" | "on-close" | "on-destroy" | "on-month-change" | "on-open" | "on-year-change" | "on-value-update" | "on-day-create" | "on-parse-config" | "on-ready" | "on-pre-calendar-position" | "on-key-down", VNodeProps & AllowedComponentProps & ComponentCustomProps, Readonly<ExtractPropTypes<{
+    modelValue: {
```
Comment (ankurk91) [inline src/component.d.ts:74]: Why do we have to repeat the props definition ?

**Comment by ankurk91** [comment]
I have to blindly accept this PR, I am not a TS expert. 

But looks like we need to convert the whole code base to TS :smile:

File: src/component.d.ts (hunk 1 above)
Comment (mimbrown) [reply src/component.d.ts:74]: Honestly... not quite sure. The typings for Vue are pretty rough, as you said, so it's easier to write a component in ts, then generate the types and go from there. So this is auto-generated then hand edited. I didn't mess with this part. We could test removing it and see if that causes any problems.

**Comment by mimbrown** [comment]
> But looks like we need to convert the whole code base to TS 😄

That would be easier from a coding perspective, but as you said it complicates the webpack/testing setup

**Comment by ankurk91** [comment]
Published a new version with beta tag

```
npm install vue-flatpickr-component@9.0.7
```

**Pull request merged**